enterprise_subsidy/apps/transaction/signals/handlers.py

```diff
@@ -1,23 +1,65 @@
 """
 Subsidy Service signals handler.
+
+The following two scenarios detail what happens when either ECS or a learner initiates unenrollment, and explains how
+infinite loops are terminated.
+
+1. When ECS invokes transaction reversal:
+=========================================
+* Reversal gets created.
+  ↳ Emit TRANSACTION_REVERSED signal.
+* TRANSACTION_REVERSED triggers the `listen_for_transaction_reversal()` handler.
+  ↳ Revoke internal & external fulfillments.
+    ↳ Emit LEARNER_CREDIT_COURSE_ENROLLMENT_REVOKED openedx event.
+  ↳ Emit LEDGER_TRANSACTION_REVERSED openedx event.
+* LEARNER_CREDIT_COURSE_ENROLLMENT_REVOKED triggers the `handle_lc_enrollment_revoked()` handler.
+  ↳ Fail first base case (reversal already exists) and quit. <-------THIS TERMINATES THE INFINITE LOOP!
+* LEDGER_TRANSACTION_REVERSED triggers the `update_assignment_status_for_reversed_transaction()` handler.
+  ↳ Updates any assignments as needed.
+
+2. When a learner invokes unenrollment:
+=======================================
+* Enterprise app will perform internal fulfillment revocation.
+  ↳ Emit LEARNER_CREDIT_COURSE_ENROLLMENT_REVOKED openedx event.
+* LEARNER_CREDIT_COURSE_ENROLLMENT_REVOKED triggers the `handle_lc_enrollment_revoked()` handler.
+  ↳ Revoke external fulfillments.
+  ↳ Create reversal.
+    ↳ Emit TRANSACTION_REVERSED signal.
+* TRANSACTION_REVERSED triggers the `listen_for_transaction_reversal()` handler.
+  ↳ Attempt to idempotently revoke external enrollment (no-op).
+  ↳ Attempt to idempotently revoke internal enrollment (no-op). <----THIS TERMINATES THE INFINITE LOOP!
```

Comment:
A call to the enterprise enrollment API still gets made here, though, right?

Reply:
Yea, what I meant by "no-op" is just that the unenroll API response was 2xx but didn't do anything because the fulfillment was already revoked. I can clarify.